src/GridParser.ts

```diff
@@ -1,60 +1,60 @@
-import { gridNotFoundErr, headerNotFoundErr, invalidDataTypeErr, invalidValueInColumnErr } from "./utils/constants";
-import { ColumnMetadata, DataTypes, Grid, TableData, TableDataParser } from "./types";
+import { gridNotFoundErr } from "./utils/constants";
+import { Grid, TableData, TableDataParser } from "./types";
 
 export default class GridParser implements TableDataParser {
     public parseToTableData(initialDataGrid: Grid): TableData | undefined {
         if (!initialDataGrid) {
             return undefined;
         }
 
-        this.validateGridHeader(initialDataGrid);
-        const rows: string[][] = this.parseGridRows(initialDataGrid, initialDataGrid.header);
+        const columnNames: string[] = this.generateColumnNames(initialDataGrid);
+        const rows: string[][] = this.parseGridRows(initialDataGrid);
 
-        return { columnMetadata: initialDataGrid.header, rows: rows };
+        return { columnNames: columnNames, rows: rows };
     }
 
-    private parseGridRows(initialDataGrid: Grid, columnMetadata: ColumnMetadata[]): string[][] {
+    private parseGridRows(initialDataGrid: Grid): string[][] {
         const gridData: (string | number | boolean)[][] = initialDataGrid.gridData;
         if (!gridData) {
             throw new Error(gridNotFoundErr);
         }
 
         const rows: string[][] = [];
-        for (const rowData of gridData) {
+        if (initialDataGrid.promoteHeaders) {
+            const row: string[] = [];
+            for (const prop in gridData[0]) {
```

Comment:
Not sure I understand this part. if promote shouldn't we just skip the first row when building the cell data?